Content/Skills/Magic/Fireball.cs

```diff
@@ -122,19 +263,482 @@ public override void OnKill(int timeLeft)
 			int area = Skill.GetTotalAreaOfEffect(160);
 			Projectile.Resize(area, area);
 			Projectile.Damage();
+
+			if (Owner.HasTreePassive<FireballTree, ScorchedEarth>())
+			{
+				int type = ModContent.ProjectileType<StickyFlame>();
+				int damage = Projectile.damage / (Owner.HasTreePassive<FireballTree, StrongerScorchedEarth>() ? 2 : 4);
+
+				for (int i = 0; i < 8; ++i)
+				{
+					Vector2 vel = Main.rand.NextVector2Circular(4, 4) + Projectile.velocity * 0.25f;
+					float timeExtension = Owner.GetPassiveStrength<FireballTree, LongerScorchedEarth>() is not 0 and int value ? 60 * value : 0;
+					int proj = Projectile.NewProjectile(Projectile.GetSource_Death(), Projectile.Center, vel, type, damage, 0, Main.myPlayer, 0, timeExtension);
+					Main.projectile[proj].frameCounter = Main.rand.Next(800);
+					Main.projectile[proj].GetGlobalProjectile<ElementalProjectile>().AddElementalValues((ElementType.Fire, 0, 1));
+				}
+			}
+
+			if (Owner.HasSkillSpecialization<Fireball, ShadowflamePyre>())
+			{
+				int type = ModContent.ProjectileType<ShadowflamePyreProjectile>();
+				float timeExtension = Owner.GetPassiveStrength<FireballTree, EverburningPyre>() is not 0 and int value ? ShadowflamePyreProjectile.MaxTimeLeft * value * 0.2f : 0;
+				Vector2 placePos = Projectile.Center;
+				Vector2 projBaseSize = ShadowflamePyreProjectile.BaseSize;
+				
+				while (Collision.SolidCollision(placePos - projBaseSize / 2, (int)projBaseSize.X, (int)projBaseSize.Y))
+				{
+					placePos.Y--;
+				}
```

Comment:
Just in case.
```suggestion
				while (Collision.SolidCollision(placePos - projBaseSize / 2, (int)projBaseSize.X, (int)projBaseSize.Y) && placePos.Y > 0)
				{
					placePos.Y--;
				}
```